src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -38,42 +38,12 @@ WindowEmperor::WindowEmperor() noexcept :
     });
 
     _dispatcher = winrt::Windows::System::DispatcherQueue::GetForCurrentThread();
-
-    // BODGY
-    //
-    // There's a mysterious crash in XAML on Windows 10 if you just let the App
-    // get dtor'd. By all accounts, it doesn't make sense. To mitigate this, we
-    // need to intentionally leak a reference to our App. Crazily, if you just
-    // let the app get cleaned up with the rest of the process when the process
-    // exits, then it doesn't crash. But if you let it get explicitly dtor'd, it
-    // absolutely will crash on exit.
-    //
-    // GH#15410 has more details.
-
-    auto a{ _app };
-    ::winrt::detach_abi(a);
 }
 
-// Disable the "destructor never returns, potential memory leak" warning - we're literally already exiting here.
-#pragma warning(suppress : 4722)
 WindowEmperor::~WindowEmperor()
 {
-    // BODGY: If the emperor is being dtor'd, it's because we've gone past the
-    // end of main, and released the ref in main. Here, we want to manually
-    // terminate our process.
-    //
-    // If you just do a
-    //
-    // _app.Close();
-    //
-    // here, then we might run into an edge case where main releases it's ref to
-    // the emperor, but one of the window threads might be in the process of
-    // exiting, and still holding a strong ref to the emperor. In that case, we
-    // can actually end up with the _window thread_ being the last reference,
-    // and calling App::Close on that thread will crash us with a E_WRONG_THREAD
-    //
-    // For more context, see MSFT:46744208
-    std::exit(0);
+    _app.Close();
+    _app = nullptr;

```

Comment:
We have tried to remove this `App` leak _two times_ and every time reintroduces a crash on shutdown. I really don't think we will succeed a third time.

Are we _totally sure_ about this?

Reply:
Yes, this will work. The destructor will never run because we terminate before the class leaves the scope.